### DISCUSSION TITLE
Issue with Subnetwork Configuration in PyPSA 0.30.0 - "Passive flows are not allowed for non-electric networks"

### DISCUSSION THREAD

**Discussion by fuch004** [PyPSA/PyPSA#1025] [General]
Hello, I am encountering an issue with subnetwork configuration while using PyPSA version 0.30.0. When I create a network with multiple buses and branches, I receive warnings that the subnetwork is not electric and that "Passive flows are not allowed for non-electric networks!" I have already set the carrier attribute to 'electricity' for all buses, generators, lines, and storage units, but the problem persists.

Problem Details

In my network:
- Number of Buses: 2
- Number of Generators: 2
- Number of Loads: 2
- Number of Storage Units: 1
- Number of Lines: 1

Despite setting all components with `carrier='electricity'`, I keep getting the warnings. Below is the main part of my code:

```python
import pypsa
import logging

# Set up logging
logging.basicConfig(level=logging.INFO, format='%(asctime)s - %(levelname)s - %(message)s')

# Create a new PyPSA network
network = pypsa.Network()

# Add electricity carrier
network.add("Carrier", "electricity")

# Add buses
network.add("Bus", "bus1", v_nom=220, carrier="electricity")
network.add("Bus", "bus2", v_nom=220, carrier="electricity")

# Add generators
network.add("Generator", "gen1", bus="bus1", p_nom=100, marginal_cost=50, carrier="electricity")
network.add("Generator", "gen2", bus="bus2", p_nom=150, marginal_cost=30, carrier="electricity")

# Add loads
network.add("Load", "load1", bus="bus1", p_set=50, carrier="electricity")
network.add("Load", "load2", bus="bus2", p_set=70, carrier="electricity")

# Add storage unit
network.add("StorageUnit", "battery", bus="bus1", p_nom=50, max_hours=2,
            efficiency_store=0.9, efficiency_dispatch=0.9, carrier="electricity")

# Add a line
network.add("Line", "line1", bus0="bus1", bus1="bus2", x=0.1, r=0.01, s_nom=100, carrier="electricity")

# Check carrier settings for all components
network.determine_network_topology()
for subnetwork in network.sub_networks.itertuples():
    if subnetwork.carrier != 'electricity':
        network.sub_networks.at[subnetwork.Index, 'carrier'] = 'electricity'

# Print subnetwork information
print("--- Subnetwork Information ---")
print(network.sub_networks)

# Optimize using linopy
try:
    network.optimize(solver_name='highs')
    print(f"Optimization completed successfully. Objective value: {network.objective}")
except Exception as e:
    logging.error(f"Optimization failed with error: {e}")
```

Current Output and Warnings

When running the above code, I receive the following warning:

```
2024-09-08 00:38:29,567 - WARNING - Warning, sub network 0 is not electric but contains multiple buses
and branches. Passive flows are not allowed for non-electric networks!
```

Even though the optimization completes successfully, the warning persists. I have tried various methods to update the subnetwork configuration, but none have resolved the issue.

Request for Help

I am looking for guidance on:
1. The correct way to set subnetworks as electric networks in PyPSA version 0.30.0.
2. Any updated documentation or instructions that might address this type of subnetwork carrier identification issue.

Thank you very much for any assistance or suggestions you can provide!

Environment Information
- PyPSA Version: 0.30.0
- Python Version: 3.11
- Operating System: Windows 11

**Comment by lkstrp** [comment]
Please do not upload any text/ docs files, just write the subject in the text box.

Duplicate of https://github.com/PyPSA/PyPSA/issues/1026.

The carrier must be AC or DC. Electricity will not get matched.
